pkg/engine/plan.go

```diff
@@ -242,18 +243,22 @@ func (res *planResult) Close() error {
 }
 
 // printPlan prints the plan's result to the plan's Options.Events stream.
-func printPlan(ctx *Context, result *planResult, dryRun bool) (ResourceChanges, error) {
-	result.Options.Events.preludeEvent(dryRun, result.Ctx.Update.GetTarget().Config)
+func printPlan(ctx *Context, planResult *planResult, dryRun bool) (ResourceChanges, *result.Result) {
+	planResult.Options.Events.preludeEvent(dryRun, planResult.Ctx.Update.GetTarget().Config)
 
 	// Walk the plan's steps and and pretty-print them out.
-	actions := newPlanActions(result.Options)
-	if err := result.Walk(ctx, actions, true); err != nil {
-		return nil, errors.New("an error occurred while advancing the preview")
+	actions := newPlanActions(planResult.Options)
+	if res := planResult.Walk(ctx, actions, true); res != nil {
+		if res.Error() == nil {
+			return nil, res
+		}
+
+		return nil, result.Error("an error occurred while advancing the preview")
```

Comment:
note: this logic is strange (and was strange before IMO).  We would previously get an error, but then drop it on the floor and return a new error.  I'm preserving that behavior, but i don't really like it.  However, if we don't have this logic, we'll end up repeating the information abotu errors we've already printed.

Reply:
open issue on killing this line and properly following pattern here.